### PR TITLE
Update Methods.md

### PR DESCRIPTION
**Pull request by matsduf**

### CODE REVIEW

File: docs/specifications/tests/Methods.md [hunk 1]
```diff
@@ -1,74 +1,135 @@
...
+   record in the answer section, the method fails, and no parent zone is found.
+6. If the recurse lookup is exhausted or ends in a loop, then the method fails
+   and no parent zone is found.
 
 
 ## Method 2: Delegation name servers
 
-Obtain the authoritative name servers for the given zone as defined in
-its delegation in the parent zone.
+Obtain the name servers (NS records) for the given zone as defined in
```
Comment (mattias-p) [inline docs/specifications/tests/Methods.md]: "name servers (NS records)" -> "NS records"?

File: docs/specifications/tests/Methods.md [hunk 2]
```diff
@@ -1,74 +1,135 @@
...
+3. Collect all IPv4 and IPv6 addresses in the input to the undelegated test and 
+   store in a cache where the connection between name server name and IP address
+   is kept. The cache is to be used by method 4.
 
 
 ## Method 3: In-zone name servers
 
-Obtain the authoritative name servers for the given zone as defined in
-the zone itself.
+Obtain the authoritative name servers (NS) for the given zone (child zone) as
```
Comment (mattias-p) [inline docs/specifications/tests/Methods.md]: "authoritative name servers (NS)" -> "names of the authoritative name servers"?

Comment (mattias-p) [inline docs/specifications/tests/Methods.md]: Is the "(child zone)" addition needed for disambiguation?

File: docs/specifications/tests/Methods.md [hunk 3]
```diff
@@ -1,74 +1,135 @@
...
-2. Look up addresses for all obtained name servers that are
-   out-of-bailiwick.
-3. Send an NS query for the given domain to all obtained name servers.
+1. Obtain name server addresses using Method 4.
+2. Send an NS query for the given zone to all obtained name server addresses.
+3. Ignore response unless AA flag is set.
 4. Record all the unique NS records in the answer sections of the
-   responses in step 3, as well as all unique glue records.
+   responses in step 2.
+5. Return all name server names (RDATA) of the NS from step 3.
```
Comment (mattias-p) [inline docs/specifications/tests/Methods.md]: "step 3" -> "step 2"?

Comment (mattias-p) [inline docs/specifications/tests/Methods.md]: What's the difference between "record" and "return"?

**Review by mattias-p — CHANGES_REQUESTED** [review]

File: docs/specifications/tests/Methods.md (hunk 2 above)
Comment (matsduf) [reply docs/specifications/tests/Methods.md]: I think it makes the risk of confusion lower.

**Review by mattias-p — APPROVED** [review]

**Pull request merged**